rice-framework/krad-data/src/main/java/org/kuali/rice/krad/data/provider/impl/CompositeMetadataProviderImpl.java

```diff
@@ -71,7 +72,7 @@ protected synchronized void initializeMetadata(Collection<Class<?>> types) {
 				// from the keys of the map
 				Map<Class<?>, DataObjectMetadata> metadata = null;
 				if (provider.requiresListOfExistingTypes()) {
-					metadata = provider.provideMetadataForTypes(masterMetadataMap.keySet());
```

Comment:
Oh, I should mention this fixes an issue that occurs when you compile with the jdk 1.8 compiler (with source and target set to 1.7) but then try to run the code on a 1.7 jdk (which Eclipse tries to do when you are running tests).